pytorch_lightning/plugins/training_type/deepspeed.py

```diff
@@ -544,21 +685,33 @@ def update_global_step(self, total_batch_idx: int, current_global_step: int) ->
             return current_global_step
 
     @classmethod
-    def register_plugins(cls, plugin_registry: Dict) -> None:
+    def register_plugins(cls, plugin_registry: 'TrainingTypePluginsRegistry') -> None:
         plugin_registry.register("deepspeed", cls, description="Default DeepSpeed Plugin")
         plugin_registry.register("deepspeed_stage_2", cls, description="DeepSpeed with ZeRO Stage 2 enabled", stage=2)
         plugin_registry.register(
             "deepspeed_stage_2_offload",
             cls,
             description="DeepSpeed ZeRO Stage 2 and CPU Offload",
             stage=2,
-            cpu_offload=True
+            offload_optimizer=True
         )
         plugin_registry.register("deepspeed_stage_3", cls, description="DeepSpeed ZeRO Stage 3", stage=3)
         plugin_registry.register(
             "deepspeed_stage_3_offload",
             cls,
             description="DeepSpeed ZeRO Stage 3 and CPU Offload",
             stage=3,
-            cpu_offload=True
+            offload_optimizer=True,
+            offload_parameters=True,
+        )
+        plugin_registry.register(
```

Comment:
That's really neat !

Reply:
Could we link this with LightningCLI, so it appears with annotation and docstring ?

Reply:
Thanks! Hopefully we can do this in a separate PR :)

Reply:
I don't think this will work automatically as LightningCLI uses the signature type for the help message.
But `plugins` in the Trainer is annotated as `str` (among others).

We would need to create an Enum with all the valid strings and use it for the typing annotation for it to work